### ISSUE TITLE
Black Screen when Attempting to Join Server Running Mod Map (Svartalfheim_WP or Forglar_WP)

### ISSUE THREAD

**Issue by rupponi** [mschnitzer/ark-survival-ascended-linux-container-image#75]
Hi mschnitzer,

I'm currently running these servers on my dedicated server at my family home, running Ubuntu 22.04 LTS with 16GB of RAM.
Other than this server, I run Plex, a Minecraft Bedrock Server, and a Palworld and Valheim server that I shut off if more resources are needed for another task.

I enjoyed running a container with The Island and, while I could not discover it in the server manager in the game, I had no problem connecting using `connect XXX.XXX.XXX.XXX` in the ARK console. There were no issues whatsoever and performance was great, allowing me to run a good playthrough of the map before I considered mod maps.

When considering mod maps, I first looked to Forglar (Map name: Forglar_WP, Mod ID: 935835). While resources were consumed (~10GB of RAM), connecting to the map yielded only a black screen. This was different to previous cases of timeouts I had received, as the application stuck to this black screen indefinitely and my GPU was even showing activity in the Task Manager.

I then chose to start with Svartalfheim given that you already provided good documentation on start up using the docker-compose file, only to receive the same conditions.

Below is the starting parameters line I used for both servers, the example is just for Svartalfheim:
```
ASA_START_PARAMS=Svartalfheim_WP?listen?Port=7777?RCONPort=27015?RCONEnabled=True -WinLiveMaxPlayers=50 -mods=893657 -NoBattlEye -clusterid=default -ClusterDirOverride="/home/gameserver/cluster-shared"
```

Below is also the standard logs of starting the server using `docker-compose start asa-server-1`:
```
sudo docker logs -f asa-server-1
Redirecting stderr to '/home/gameserver/Steam/logs/stderr.txt'
Logging directory: '/home/gameserver/Steam/logs'
[  0%] Checking for available updates...
[----] Verifying installation...
UpdateUI: skip show logoSteam Console Client (c) Valve Corporation - version 1718904263
-- type 'quit' to exit --
Loading Steam API...OK

Connecting anonymously to Steam Public...OK
Waiting for client config...OK
Waiting for user info...OK
 Update state (0x5) verifying install, progress: 0.02 (2323326 / 12334487960)
 Update state (0x5) verifying install, progress: 5.05 (622866624 / 12334487960)
 Update state (0x5) verifying install, progress: 13.17 (1624256704 / 12334487960)
 Update state (0x5) verifying install, progress: 21.16 (2609702383 / 12334487960)
 Update state (0x5) verifying install, progress: 29.23 (3605141359 / 12334487960)
 Update state (0x5) verifying install, progress: 37.28 (4598561487 / 12334487960)
 Update state (0x5) verifying install, progress: 45.39 (5598238335 / 12334487960)
 Update state (0x5) verifying install, progress: 53.47 (6595225059 / 12334487960)
 Update state (0x5) verifying install, progress: 61.15 (7542941831 / 12334487960)
 Update state (0x5) verifying install, progress: 69.10 (8523607255 / 12334487960)
 Update state (0x5) verifying install, progress: 77.20 (9521876521 / 12334487960)
 Update state (0x5) verifying install, progress: 85.29 (10520155737 / 12334487960)
 Update state (0x5) verifying install, progress: 93.40 (11520249401 / 12334487960)
Success! App '2430930' fully installed.
Starting the ARK: Survival Ascended dedicated server...
Start parameters: Svartalfheim_WP?listen?Port=7777?RCONPort=27015?RCONEnabled=True -WinLiveMaxPlayers=50 -mods=893657 -NoBattlEye -clusterid=default -ClusterDirOverride="/home/gameserver/cluster-shared"
esync: up and running.
Failed to read /etc/machine-id, error Invalid argument.
wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
```

Would be curious if there's any known resolution as to why the client blacks the screen but doesn't timeout when using mod maps with the server. Please let me know if there's any more context I can provide and thanks for your help as well for all your work with this ASA dedicated Linux server solution!

**Comment by mschnitzer** [comment]
I'm in the middle of a move, I'm coming back to you mid next week.

**Comment by rupponi** [comment]
No problem! Just went through a move myself last week, take your time and thanks for the response!

**Comment by rupponi** [comment]
Hey Manuel! Hope the move was great, I also had to put setting up a server on pause for some time, but was wondering if there was any updates on fixes or ways to avoid the black screen issue when running the Linux server with non-standard maps?
Also for reference, I had no issue running TheIsland_WP and was able to run a session on there without any interruptions, crashes, or data corruption.
Based on the memory consumption, I'm assuming the container does successfully load the map, but connecting simply leads to the black screen issue this thread is about.
If you have any thoughts or testing methodologies that I could use to debug this, that'd be great! Thanks!

**Comment by mschnitzer** [comment]
Hi, I'm coming back to you this weekend.

**Comment by mschnitzer** [comment]
I suspect that connecting directly to servers with mod maps won't work, because the client would then not ask you whether you want to download the mods before joining. I'd suggest you try connecting through the server browsers, by making sure you followed [these steps](https://github.com/mschnitzer/ark-survival-ascended-linux-container-image?tab=readme-ov-file#server-is-not-visible-in-server-browser) and report back whether that works. Otherwise I could imagine that installing the mods manually (somehow) on your client would help as well.

**Comment by rupponi** [comment]
> I suspect that connecting directly to servers with mod maps won't work, because the client would then not ask you whether you want to download the mods before joining. I'd suggest you try connecting through the server browsers, by making sure you followed [these steps](https://github.com/mschnitzer/ark-survival-ascended-linux-container-image?tab=readme-ov-file#server-is-not-visible-in-server-browser) and report back whether that works. Otherwise I could imagine that installing the mods manually (somehow) on your client would help as well.

Gotcha, yeah I'll try a few approaches. If I have any success, I'll leave my approach in another comment.

**Comment by rupponi** [comment]
Ended up resolving things, turns out the original configuration I ran, for some reason, was affecting new containers I was standing up even with a re-run of the compose. I did a clearing of the server data in /var/lib/docker and re-ran.

Server manager in ARK was also being weird but this last time it recognized the server in the manager. Now I can consistently get the server running with Svartalfheim.

Closing out this issue.
